packages/cli/src/lib/updateCheck.js

```diff
@@ -0,0 +1,313 @@
+import { spawn } from 'child_process'
+import fs from 'fs'
+import os from 'os'
+import path from 'path'
+
+import boxen from 'boxen'
+import chalk from 'chalk'
+import latestVersion from 'latest-version'
+import semver from 'semver'
+
+import { getConfig } from '@redwoodjs/internal/dist/config'
+
+import { setLock, unsetLock } from './locking'
+
+import { getPaths } from './index'
+
+/**
+ * @typedef {{
+ *   localVersion: string,
+ *   remoteVersions: Map<string, string>,
+ *   checkedAt: number,
+ *   shownAt: number,
+ * }} UpdateData
+ */
+
+/**
+ * @const {number} The number of milliseconds between update checks (24 hours)
+ */
+const CHECK_PERIOD = 24 * 60 * 60_000
+
+/**
+ * @const {number} The number of milliseconds between showing a user an update notification (24 hours)
+ */
+const SHOW_PERIOD = 24 * 60 * 60_000
+
+/**
+ * @const {number} The default datetime for shownAt and checkedAt in milliseconds, corresponds to 2000-01-01T00:00:00.000Z
+ */
+export const DEFAULT_DATETIME_MS = 946684800000
+
+/**
+ * @const {string} The identifier used for the lock within the check function
+ */
+export const LOCK_IDENTIFIER = 'UPDATE_CHECK'
+
+/**
+ * @const {string[]} The name of commands which should NOT execute the update checker
+ */
+export const EXCLUDED_COMMANDS = ['upgrade', 'ts-to-js']
+
+/**
+ * @const {string} Filepath of the file which persists update check data within the .redwood directory
+ */
+export const PERSISTENCE_DIRECTORY = path.join(
+  getPaths().generated.base,
+  'updateCheck'
+)
```

Comment:
Here's the problem with the cwd test—`getPaths` here runs as soon as this file is imported, which happens before we set the cwd in the CLI's index.js file

Reply:
Here's the fix @Josh-Walker-GM https://github.com/redwoodjs/redwood/pull/7162/commits/e1c257f816f40d71306e715f75e44b34ec8860ae, feel free to rework the logic if you want

Reply:
Thanks for clearing this up, appreciate it. I'm happy with the way you've done it.